web/models/user_from_auth.ex

```diff
@@ -94,24 +94,28 @@ defmodule Parma.UserFromAuth do
     end
   end
 
-  defp create_user_from_auth(auth, current_user, repo) do
-    user = current_user
-    if !user, do: user = repo.get_by(User, email: auth.info.email)
-    if !user do
-      user = create_user(auth, repo)
-      Toniq.enqueue(UserInitialization, user: user)
-    end
+  defp create_user_from_auth(auth, current_user, repo) do    
+    user = current_user || repo.get_by(User, email: auth.info.email)    
+    user = 
+      case user do
+        nil -> create_user(auth, repo)
+        _ -> user
+      end    
     authorization_from_auth(user, auth, repo)
     {:ok, user}
   end
 
   defp create_user(auth, repo) do
     name = name_from_auth(auth)
-    result = User.registration_changeset(%User{}, scrub(%{email: auth.info.email, name: name}))
-    |> repo.insert
+    result = 
+      User.registration_changeset(%User{}, scrub(%{email: auth.info.email, name: name}))
+      |> repo.insert
     case result do
-      {:ok, user} -> user
-      {:error, reason} -> repo.rollback(reason)
+      {:ok, user} -> 
+        Toniq.enqueue(UserInitialization, user: user)
+        user
+      {:error, reason} -> 
+        repo.rollback(reason)
```

Comment:
so we have to manually rollback?

Reply:
Yes, unless in an Ecto transaction or Multi, we need to rollback manually.